climate_learn/models/components/resnet.py

```diff
@@ -206,15 +225,19 @@ def test_rollout(self, x: torch.Tensor, y: torch.Tensor, clim, variables, out_va
         if steps > 1:
             assert len(variables) == len(out_variables)
 
-        preds = []
-        for _ in range(steps):
-            x = self.predict(x)
-            if self.prob_type == 'parametric':
-                x = mean_transform(x.loc)
-            preds.append(x)
-        preds = torch.stack(preds, dim=1)
-        if len(y.shape) == 4:
-            y = y.unsqueeze(1)
+        # need debug here!! what to do with categorical and other prob_type?
```

Comment:
Can you elaborate on this? What is the issue?

Reply:
So the original code for this part was everything inside the else statement (from line 232 to 240) and when I ran the categorical, errors popped up, and I wasn't able to fix them at that time, so I made an if statement and simply ran`preds = self.predict(x, len(out_variables))` for categorical. 

And I'm bit confused why did the val_rollout has different implementations for probability type and regular deterministic methods. while for test rollout, the original code doesn't have different implementations?

Reply:
It doesn't look like we have a good reason for why val_rollout is different from test_rollout. I dug through our commit history and it looks like they just ended up that way because of the uncertainty repo merge: https://github.com/aditya-grover/climate-learn/blob/300e602b06072c3d9e8e19841c9526998debb3c7/climate_learn/models/components/resnet.py. 

So IMO we have the flexibility to change the rollouts to match each other. I think it makes sense to have them be the same. Can you see what happens if you make the rollouts match? Does our demo notebook still run?

Reply:
Looks like it's working

Reply:
Great, thanks! Once you resolve the merge conflicts, I will approve this PR.